ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java

```diff
@@ -234,6 +234,10 @@ private static void ensureMax(final BigDecimal value, final int precision, final
   }
 
   public static SqlType fromValue(final BigDecimal value) {
+    final BigDecimal bigDecimalZero = new BigDecimal("0");
+    if (value.compareTo(bigDecimalZero) == 0) {
```

Comment:
let's add a comment in here explaining what's going on on and why we have to do this